integration/benchmarks/benchmarks_analysis.ipynb

```diff
@@ -0,0 +1,607 @@
+    "    bars1 = ax1.bar(x, ig_process_cpu, alpha=0.8, color='orange',\n",
+    "                    yerr=ig_process_cpu_ci, capsize=5, label='IG Process CPU')\n",
+    "\n",
+    "    ax1.set_xlabel('Events per Second')\n",
+    "    ax1.set_ylabel('CPU Usage (%)')\n",
+    "    ax1.set_title('CPU Usage (with 95% CI)')\n",
+    "    ax1.set_xticks(x)\n",
+    "    ax1.set_xticklabels(rps_values)\n",
+    "    ax1.grid(True, alpha=0.3, axis='y')\n",
+    "\n",
+    "    # Set dynamic Y-axis range for CPU\n",
+    "    max_cpu = np.max(ig_process_cpu)\n",
+    "    ax1.set_ylim(0, max_cpu * 1.25)\n",
+    "\n",
+    "    # Add value labels on CPU bars\n",
+    "    for i, bar in enumerate(bars1):\n",
+    "        height = bar.get_height()\n",
+    "        ci_value = ig_process_cpu_ci[i]\n",
+    "        ax1.text(bar.get_x() + bar.get_width()/2., height + ci_value,\n",
+    "                 f'{height:.2f}', ha='center', va='bottom', fontsize=9, rotation=0)\n",
+    "\n",
+    "    # Plot 2: IG Process Memory Usage\n",
+    "    bars2 = ax2.bar(x, ig_process_mem, alpha=0.8, color='red',\n",
+    "                    yerr=ig_process_mem_ci, capsize=5, label='IG Process Memory')\n",
+    "\n",
+    "    ax2.set_xlabel('Events per Second')\n",
+    "    ax2.set_ylabel('Memory Usage (MB)')\n",
+    "    ax2.set_title('Memory Usage (with 95% CI)')\n",
+    "    ax2.set_xticks(x)\n",
+    "    ax2.set_xticklabels(rps_values)\n",
+    "    ax2.grid(True, alpha=0.3, axis='y')\n",
+    "\n",
+    "    # Set dynamic Y-axis range for memory\n",
+    "    max_mem = np.max(ig_process_mem)\n",
+    "    min_mem = np.min(ig_process_mem)\n",
+    "    if max_mem > 0:\n",
+    "        ax2.set_ylim(0.75*min_mem, 1.25*max_mem)\n",
+    "    else:\n",
+    "        ax2.set_ylim(0, 10)\n",
+    "\n",
+    "    # Add value labels on Memory bars\n",
+    "    for i, bar in enumerate(bars2):\n",
+    "        height = bar.get_height()\n",
+    "        ci_value = ig_process_mem_ci[i]\n",
+    "        ax2.text(bar.get_x() + bar.get_width()/2., height + ci_value,\n",
+    "                 f'{height:.0f}', ha='center', va='bottom', fontsize=9, rotation=0)\n",
+    "\n",
+    "    # Set main title\n",
+    "    fig.suptitle(f'{gadget} - IG Process', fontsize=16, fontweight='bold')\n",
+    "\n",
+    "    plt.tight_layout()\n",
+    "\n",
+    "    return fig"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5b40d4b1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_performance_plots(filepath):\n",
+    "    \"\"\"Create CPU and memory performance plots for a single test result file\"\"\"\n",
+    "    # Read the CSV file, ignoring lines starting with #\n",
+    "    df = pd.read_csv(filepath, comment='#')\n",
+    "\n",
+    "    # Clean column names (remove extra spaces)\n",
+    "    df.columns = df.columns.str.strip()\n",
+    "\n",
+    "    # Get unique gadget names\n",
```

Comment:
When running a test with different variant of same gadget, `GadgetName` isn't unique anymore. Perhaps this should be `TestName` ?